parcels/fieldset.py

```diff
@@ -274,39 +276,56 @@ def from_netcdf(cls, filenames, variables, dimensions, indices=None, fieldtype=N
             for procvar, _ in fields.items():
                 procdims = dimensions[procvar] if procvar in dimensions else dimensions
                 procinds = indices[procvar] if (indices and procvar in indices) else indices
                 procpaths = filenames[procvar] if isinstance(filenames, dict) and procvar in filenames else filenames
                 nowpaths = filenames[var] if isinstance(filenames, dict) and var in filenames else filenames
-                if procdims == dims and procinds == inds and procpaths == nowpaths:
-                    sameGrid = False
+                if procdims == dims and procinds == inds:
+                    processedGrid = False
                     if ((not isinstance(filenames, dict)) or filenames[procvar] == filenames[var]):
-                        sameGrid = True
+                        processedGrid = True
                     elif isinstance(filenames[procvar], dict):
-                        sameGrid = True
+                        processedGrid = True
                         for dim in ['lon', 'lat', 'depth']:
                             if dim in dimensions:
-                                sameGrid *= filenames[procvar][dim] == filenames[var][dim]
-                    if sameGrid:
+                                processedGrid *= filenames[procvar][dim] == filenames[var][dim]
+                    if processedGrid:
                         grid = fields[procvar].grid
-                        kwargs['dataFiles'] = fields[procvar].dataFiles
-                        break
+                        # ==== check here that the dims of field_chunksize are the same ==== #
+                        if grid.master_chunksize is not None:
+                            res = False
+                            if (isinstance(chunksize, tuple) and isinstance(grid.master_chunksize, tuple)) or (isinstance(chunksize, dict) and isinstance(grid.master_chunksize, dict)):
+                                res |= functools.reduce(lambda i, j: i and j, map(lambda m, k: m == k, chunksize, grid.master_chunksize), True)
+                            else:
+                                res |= (chunksize == grid.master_chunksize)
+                            if res:
+                                grid_chunksize = grid.master_chunksize
+                                logger.warning("Trying to initialize a shared grid with different chunking sizes - action prohibited. Replacing requested field_chunksize with grid's master chunksize.")
+                            else:
+                                raise ValueError("Conflict between grids of the same fieldset chunksize and requested field chunksize as well as the chunked name dimensions - Please apply the same chunksize to all fields in a shared grid!")
+                        if procpaths == nowpaths:
+                            dFiles = fields[procvar].dataFiles
+                            break
```

Comment:
Is this the best place to do this check? What if a user first creates a `FieldSet.from_netcdf()` and then later adds a `Field` via `FieldSet.add_field()`? This check then wouldn't be done. 
Would it make sense to move/combine this check with the new one in gridset.py, which is always called when a new `Field` is added to a `FieldSet`?

Reply:
I think that is so and so, can be argued. At this place, we know we treat NetCDF file data, so their grids need to have ``field_chunksize`` parameters. We also know which other fields and grids are there.

In the ``add_field()`` function we also catch later-added Fields, but there, we have the messy task to check if that Field is a netcdf field, if it has a field_chunksize, etc.

I'll have a look how to integrate this as nicely as possible.

Reply:
working on it

Reply:
the change is now done, as discussed